lib/oppen/printer.rb

```diff
@@ -249,7 +262,19 @@ def advance_right
     def advance_left(token, token_width)
       return if token_width.negative?
 
-      print_stack.print token, token_width
+      trim_on_break =
+        if token.is_a?(Token::Break)
+          # Find the first previous String token.
+          idx = (left - 1) % scan_stack.length
+          idx = (idx - 1) % scan_stack.length \
+            while idx != right && tokens[idx] && !tokens[idx].is_a?(Token::String) \
+                  && !tokens[idx].is_a?(Token::Break)
```

Comment:
Please do not use post-fix while in Ruby.  It was a bad idea.  Even Matz regrets it.

Reply:
This line is still a work in progress. Firas and I are trying to rewrite it in order to make it Ruby idiomatic.